### ISSUE TITLE
Possibility to add lock information to PROPFIND response

### ISSUE THREAD

**Issue by michalhorcic** [danmunn/redmine_dmsf#1096]
When asking webdav for file information with PROPFIND is it possible to include information about existing lock in XML response? 

My goal is to be able to display lock information in webdav clients (like WinSCP).

With some hacky changes in the code, I am somewhat able to add it there myself, but before going deeper into the code and changing things that I am not expert on. I want to ask if there is a reason why lock information is not included in PROPFIND response or if anybody was dealing with similar issue.

**Comment by picman** [comment]
Yes, it seems that the response to a PROPFIND request  misses _lockdiscovery_ field. I think that the part related to _lockdiscovery_ shoul be copied from _lock_ method to _propfind_ in _lib/dav4rack/controller.rb_. But _dav4rack_ is an external library copied from https://github.com/planio-gmbh/dav4rack an locally stored in this plugin. The right way would be to inherit _Controller_ class in DMSF the same way as e.g. _ResourceProxy_ class and write there a new implementation of _propfind_ method. Try it and make a pull request here.
